mlflow/store/artifact/s3_artifact_repo.py

```diff
@@ -80,7 +80,7 @@ def _cached_get_s3_client(
 
 
 def _get_s3_client(
-    addressing_style="path",
+    addressing_style=None,
```

Comment:
In this method we should look up addressing style from the env variable if none is specified by the caller